src/pruner_util.cpp

```diff
@@ -91,6 +91,55 @@ int get_exit_code(cstring name, P4PRUNER::PrunerConfig pruner_conf) {
     return WEXITSTATUS(system(command.c_str()));
 }
 
+cstring get_error_string(cstring name, P4PRUNER::PrunerConfig pruner_conf) {
+    cstring command = pruner_conf.compiler;
+    command += " ";
+    command += name;
+    // Apparently popen doesn't like stderr hence redirecting stderr to stdout
+    command += " 2>&1";
+
+    char buffer[128];
+    cstring result = "";
+    FILE *pipe = popen(command, "r");
+
+    try {
+        while (fgets(buffer, sizeof buffer, pipe) != NULL) {
+            result += buffer;
+        }
+    } catch (...) {
+        pclose(pipe);
+        throw;
+    }
+    pclose(pipe);
+    return result;
+}
+
+bool is_crash_bug(cstring name, P4PRUNER::PrunerConfig pruner_conf) {
```

Comment:
This is probably best expressed in the form of an enum, something like:
```cpp
enum class ErrorType : uint32_t {
  SemanticBug = 0,
  CrashBug = 1,
  Error = 2,
  Unknown = 3
}
```
Also ideally we want to retrieve the error string too and store it in the config. Maybe the function is better named as "classify_bug".